[zdn][new] 337301-oge2024

Zer00Player opened the pull request.

Zer00Player committed:
[zdn][new] 337301-oge2024

Zer00Player committed:
change coordAxis_prepare

nickkolok commented on zdn/matoge2024/7/337301.js:
```diff
@@ -0,0 +1,66 @@
+		let outsideLeft = left - 0.5;
+		let outsideRight = right + 0.5;
+		let randPointA = sl1();
+		let pointA = Math.random() * [(left - outsideLeft), (outsideRight - right)][randPointA] + [outsideLeft, right][randPointA];
```
Хотелось бы видеть `sl()`, где это возможно. Да, в JavaScript нет (вроде бы до сих пор!) нормального встроенного генератора случайных чисел с заданным шагом в заданном диапазоне, поэтому приходится использовать наш любимый велосипед. Но он, как мне кажется, читабельнее!

nickkolok commented on zdn/matoge2024/7/337301.js:
```diff
@@ -0,0 +1,66 @@
+			let x2 = mid + 75;
+			let ratio = (x2 - x1) / (borders[1] - borders[0]);
+
+			coordAxis_drawMarkPoint(ct, x1, borders[0].toString(), "line", "underAxis");
```
`.toString()`  (а лучше - неявное преобразование через сложение с пустой строкой) должно быть где-то внутри `coordAxis_drawMarkPoint` (если оно вообще требуется - может, и без него сработает, всё-таки в JS неявная типизация)

nickkolok commented on zdn/matoge2024/7/337301.js:
```diff
@@ -0,0 +1,66 @@
+		let correct = correctOptions.length === 1 ? correctOptions[0][0] : "нет данных";
+
+		let allLabels = options.map(x => x[0]);
+		if (!allLabels.includes("нет данных")) allLabels.push("нет данных");
```
Теоретически здесь можно применить `.pushIf()`, но, кажется, назрело написание функции `.pushUnique()` ? @Zer00Player , а получится ли посмотреть, как реализован `.pushIf()`, и рядышком там же сделать `.pushUnique()` ?

Zer00Player replied:
.pushUnique() не выдало ни одного результата.

nickkolok replied:
Разумеется, не выдало. Потому что этой функции пока нет. А хотелось, хотелось бы, чтобы была...

nickkolok commented on zdn/matoge2024/7/337301.js:
```diff
@@ -0,0 +1,66 @@
+		let options = [[["a", pointA], ["a^2", pointA2], ["a^3", pointA3]], [["a^2", pointA2], ["a^3", pointA3], ["a^4", pointA4]]][randA];
+
+		let isMaxTaskOrMin = sl1();
+		let targetValue = [Math.max(...options.map(x => x[1])), Math.min(...options.map(x => x[1]))][isMaxTaskOrMin];
```
`options.T()[1].minE()` ?

Zer00Player committed:
-fix #2455
change in pointA;
remove toString;
add .T().minE;
change push to pushIf;
width now 400;

SugarHedgehog commented on zdn/matoge2024/7/337301.js:
```diff
@@ -0,0 +1,66 @@
+			coordAxis_drawMarkPoint(ct, pointA_coord, "a", "dot", "overAxis");
+		};
+
+		let pointA2 = pointA ** 2;
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Exponentiation_assignment

SugarHedgehog replied:
Но возможно в этом есть смысл...но дальше pointA и др больше не используются в чистом виде

SugarHedgehog reviewed (approved):
мне не к чему придраться

nickkolok committed:
[zdn] Use coordAxis_drawAuto

nickkolok committed:
[zdn] Use LaTeX

nickkolok committed:
[zdn] Remove empty EOF rows

nickkolok committed:
337301 - the dot can belong to the interval but cannot collide with integer mark

Zer00Player committed:
Merge pull request #2 from nickkolok/newtask-337301

[zdn] Newtask 337301 - fixes

nickkolok commented on zdn/matoge2024/7/337301.js:
```diff
@@ -0,0 +1,60 @@
+
+		let outsideLeft = left - 0.5;
+		let outsideRight = right + 0.5;
+		let randPointA = sl1();
```
Эта переменная далее где-то используется?

Zer00Player committed:
remove randPointA

nickkolok committed:
Merge branch 'devel' into newtask-337301
